packages/medusa-payment-stripe/src/services/stripe-ideal.js

```diff
@@ -41,59 +34,118 @@ class IdealProviderService extends PaymentService {
-   * @param {object} paymentData - payment method data from cart
-   * @returns {string} the status of the payment intent
+   * @param {PaymentSessionData} paymentSessionData - payment method data from cart
+   * @return {Promise<PaymentSessionStatus>} the status of the payment intent
    */
-  async getStatus(paymentData) {
-    return await this.stripeProviderService_.getStatus(paymentData)
+  async getStatus(paymentSessionData) {
+    const { id } = paymentSessionData
+    const paymentIntent = await this.stripe_.paymentIntents.retrieve(id)
+
+    let status = PaymentSessionStatus.PENDING
+
+    if (paymentIntent.status === "requires_payment_method") {
+      return status
+    }
+
+    if (paymentIntent.status === "requires_confirmation") {
+      return status
+    }
+
+    if (paymentIntent.status === "processing") {
+      return status
+    }
+
+    if (paymentIntent.status === "requires_action") {
+      status = PaymentSessionStatus.REQUIRES_MORE
+    }
+
+    if (paymentIntent.status === "canceled") {
+      status = PaymentSessionStatus.CANCELED
+    }
+
+    if (paymentIntent.status === "requires_capture") {
+      status = PaymentSessionStatus.AUTHORIZED
+    }
+
+    if (paymentIntent.status === "succeeded") {
+      status = PaymentSessionStatus.AUTHORIZED
+    }
+
+    return status
   }
 
   /**
    * Fetches a customers saved payment methods if registered in Stripe.
-   * @param {object} customer - customer to fetch saved cards for
-   * @returns {Promise<Array<object>>} saved payments methods
+   * @param {Customer} customer - customer to fetch saved cards for
+   * @return {Promise<Data[]>} saved payments methods
    */
   async retrieveSavedMethods(customer) {
+    if (customer.metadata && customer.metadata.stripe_id) {
+      const methods = await this.stripe_.paymentMethods.list({
+        customer: customer.metadata.stripe_id,
+        type: "card",
+      })
+
+      return methods.data
+    }
+
     return Promise.resolve([])
   }
 
   /**
    * Fetches a Stripe customer
    * @param {string} customerId - Stripe customer id
-   * @returns {Promise<object>} Stripe customer
+   * @return {Promise<object>} Stripe customer
    */
   async retrieveCustomer(customerId) {
-    return await this.stripeProviderService_.retrieveCustomer(customerId)
+    if (!customerId) {
+      return Promise.resolve()
+    }
+    return await this.stripe_.customers.retrieve(customerId)
   }
 
   /**
    * Creates a Stripe customer using a Medusa customer.
    * @param {object} customer - Customer data from Medusa
-   * @returns {Promise<object>} Stripe customer
+   * @return {Promise<object>} Stripe customer
    */
   async createCustomer(customer) {
-    return await this.stripeProviderService_.createCustomer(customer)
+    try {
+      const stripeCustomer = await this.stripe_.customers.create({
+        email: customer.email,
+      })
+
+      if (customer.id) {
+        await this.customerService_.update(customer.id, {
```

Comment:
@srindom I think that we should pass the transaction here as well. wdyt?
Probably that we should also check the webhooks?